### PR TITLE
Add `animation-tick()` builtin function

### PR DESCRIPTION
**Pull request by ogoffart**
Low level primitive allowing to build always running animations

### CODE REVIEW

**Comment by ogoffart** [comment]
> Any way to test this, at least compile time wise? :)

Looks like I forgot to git add the test

**Comment by ogoffart** [comment]
I ammended the commit with the test and your suggestion

**Comment by tronical** [comment]
> I ammended the commit with the test and your suggestion

👍

**Comment by tronical** [comment]
Ah, ChangeLog is missing :)

**Comment by ogoffart** [comment]
Added changelog and made `mod` a macro

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,9 @@ All notable changes to this project are documented in this file.
 
 ## Unreleased
 
+### Changed
+ - `mod` now work on more types than just integer, it can work on any numeric type
```
Comment (tronical) [inline CHANGELOG.md:7]:
```suggestion
 - `mod` now works on any numeric type, not only integers.
```

File: docs/langref.md [hunk 2]
```diff
@@ -1055,6 +1055,30 @@ the focus the very first time the window receives the focus - it becomes the ini
 
 The debug function take a string as an argument and prints it
 
+* **`animation-tick() -> duration`**:  This function returns a monotonically increasing time, which can be used for animations.
+    Calling this function from a binding will constantly re-evaluate the binding.
+    It can be used like so: `x: 1000px + sin(animation-tick() / 1s * 360deg) * 100px;` or `y: 20px * mod(animation-tick(), 2s) / 2s `
+
+```
+Example := Window {
```
Comment (tronical) [inline docs/langref.md:1063]: Nice :)

**Review by tronical — APPROVED** [review]

**Pull request merged**